formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonToStringWriter.kt

```diff
@@ -0,0 +1,10 @@
+package kotlinx.serialization.json.internal
+
+internal expect class JsonToStringWriter constructor() : JsonWriter {
+    override fun writeChar(char: Char)
+    override fun writeLong(value: Long)
+    override fun write(text: String)
+    override fun writeQuoted(text: String)
```

Comment:
It seems a little double up to put the semantics of writing the different parts of json here, where every implementation has to re-implement the same translation semantics (I know that it generally is the same as the platform `.toString()`)

Reply:
In this case, only `writeLong` looks redundant. `Long.toString` is used here, but this is not the optimal solution in JVM, because such an implementation internally creates a temporary buffer for characters and creates a new `String` instance. In the case of small classes (for example, `data class Foo(var i: Int)`), array allocation can greatly slow down performance.
This can be optimized, but within a different task.

The implementations of other functions may differ significantly for different platforms and writers, so it is not optimal to put them in the general code.